rotkehlchen/chain/evm/decoding/thegraph/balances.py

```diff
@@ -64,71 +67,19 @@ def _query_vesting_balances(self, balances: 'BalancesSheetType') -> 'BalancesShe
-                continue
-
-            user_address = string_to_evm_address(event.location_label)
-            contract = EvmContract(
-                address=string_to_evm_address(delegator),
-                abi=GRAPH_TOKEN_LOCK_WALLET_ABI,
-                deployed_block=0,
-            )
-            encoded_call = contract.encode(method_name='totalOutstandingAmount')
-            vesting_contract_calls.append((contract.address, encoded_call))
-            call_mapping.append((user_address, contract))
-
-        if len(vesting_contract_calls) == 0:
-            return balances
-
-        try:
-            outputs = self.evm_inquirer.multicall_2(require_success=False, calls=vesting_contract_calls)  # noqa: E501
-        except (RemoteError, BlockchainQueryError) as e:
-            log.error(f'Failed to query GRT vested balance due to {e!s}')
-            return balances
-
-        results: list[tuple[ChecksumEvmAddress, int]] = [
-            (address, delegator.decode(result=result[1], method_name='totalOutstandingAmount')[0])
-            for result, (address, delegator) in zip(outputs, call_mapping, strict=True)
-            if result[0] is True and len(result[1]) != 0
-        ]
-        grt_price = Inquirer.find_usd_price(self.token)
-        for address, balance in results:
-            balance_norm = token_normalized_value(token_amount=balance, token=self.token)
-            balances[address].assets[self.token] += Balance(
-                amount=balance_norm,
-                usd_value=grt_price * balance_norm,
-            )
-
-        return balances
-
-    def query_balances(self) -> 'BalancesSheetType':
-        """
-        Query balances of GRT tokens delegated to indexers if deposit events are found.
-        First, the current shares amounts are fetched from the contract,
-        then shares are converted into GRT balances according to the current pool ratio.
-        The results include delegation rewards earned over time.
-        """
-        balances: BalancesSheetType = defaultdict(BalanceSheet)
-        balances = self._query_vesting_balances(balances)
-
-        # fetch deposit events
-        addresses_with_deposits = self.addresses_with_deposits(products=[EvmProduct.STAKING])
         # remap all events into a list that will contain all pairs (delegator, indexer)
         delegations_unique = set()
-        for delegator, event_list in addresses_with_deposits.items():
-            for event in event_list:
-                if event.extra_data is None:
-                    continue
-                if (indexer := event.extra_data.get('indexer')) is not None:
-                    delegations_unique.add((delegator, indexer))
+        for event in events:
+            if event.location_label is None or event.extra_data is None:
+                continue
+            if (indexer := event.extra_data.get('indexer')) is not None:
```

Comment:
I don't see it clear in the diff. You have here both L1 and L2 balances? Wouldn't it make sense to have two classes one for ethereum and one for arbitrum with the common parts abstracted?